src/exifmodel.cpp

```diff
@@ -0,0 +1,110 @@
+#include "exifmodel.h"
+#include <QDebug>
+#include <QLibrary>
+
+static void parseContent(ExifContent *content, void *userdata);
+static void parseEntry(ExifEntry *entry, void *userdata);
+
+typedef void (*ExifForEachEntry)(ExifContent*, ExifContentForeachEntryFunc, void*);
+typedef void (*ExifGetValue)(ExifEntry*, char*, int);
+typedef ExifIfd (*ExifGetIfd)(ExifContent*);
+typedef char* (*ExifGetTag)(ExifTag, ExifIfd);
+static ExifForEachEntry f_foreachentry;
+static ExifGetValue f_getvalue;
+static ExifGetIfd f_getifd;
+static ExifGetTag f_gettag;
+
+ExifModel::ExifModel()
+{
+}
+
+ExifModel::~ExifModel()
+{
+    free(m_data);
+    if (m_lib.isLoaded()) {
+        m_lib.unload();
+    }
+}
+
+bool ExifModel::loadLibexif()
+{
+    if (m_lib.isLoaded())
+        return true;
+
+    m_lib.setFileName("libexif.so.12");
+    if (m_lib.load())
+    {
+        f_fromfile =  (ExifFromFile) m_lib.resolve("exif_data_new_from_file");
+        f_getvalue = (ExifGetValue) m_lib.resolve("exif_entry_get_value");
+        f_foreachentry = (ExifForEachEntry) m_lib.resolve("exif_content_foreach_entry");
+        f_foreachcontent = (ExifForEachContent) m_lib.resolve("exif_data_foreach_content");
+        f_getifd = (ExifGetIfd) m_lib.resolve("exif_content_get_ifd");
+        f_gettag = (ExifGetTag) m_lib.resolve("exif_tag_get_name_in_ifd");
+        f_dump = (ExifDump) m_lib.resolve("exif_data_dump");
```

Comment:
why so complicated? would not be better to link binary with libexif and use functions directly?